packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -229,10 +229,20 @@ class MDnsVmServiceDiscovery {
       final Set<String> uniqueDomainNamesInResults = <String>{};
 
       // Listen for mDNS connections until timeout.
-      final Stream<PtrResourceRecord> ptrResourceStream = client.lookup<PtrResourceRecord>(
-        ResourceRecordQuery.serverPointer(dartVmServiceName),
-        timeout: timeout
-      );
+      final Stream<PtrResourceRecord> ptrResourceStream;
+
+      try {
+        ptrResourceStream = client.lookup<PtrResourceRecord>(
+            ResourceRecordQuery.serverPointer(dartVmServiceName),
+            timeout: timeout
+        );
+      } on SocketException {
+        throwToolExit('You may be having a permissions issue with your IDE. '
```

Comment:
Should we only output this error message on macOS?

Reply:
done